### PR TITLE
Update to HowToBuild with XCode 10.2

### PR DESCRIPTION
**Pull request by ShanghaiTimes**
The general instructions in the mac_build folder in the file HowToBuildBOINC_XCode.pdf should also note that if you want to build using XCode alone, you need to put all the downloaded libraries in the folder directly above the build folder. For EG:
All of these external programs:

wxWidgets-3.1.0 s-3.1.0.tar.bz2
sqlite-3.22.0
curl-7.58.0:
c-ares-1.13.0 (used by curl):
openssl-1.1.0g
freetype-2.9
ftgl-2.1.3~rc5

Need to go into the top GitHub directory on your system and be unpacked there. Otherwise XCode  as the project is currently set up, will not find the libraries. They need to be unzipped there of course.

/Users/<your name>/GitHub/

So that when you build in XCode, they can be found in the existing setup, which looks for them in - for example -
USER_HEADER_SEARCH_PATHS = ../../curl-7.58.0/include ../../openssl-1.1.0g/include ../lib/**

Fixes #

**Description of the Change**

**Alternate Designs**

**Release Notes**

### CODE REVIEW

**Comment by CharlieFenton** [comment]
The  _HowToBuildBOINC_XCode.pdf_ file already does say this:

>[1] Make sure you are logged into the Mac using an account with administrator privileges.  Create a parent directory within which to work.  In this description; we will call it BOINC_dev, but you can name it anything you wish.

>[2] Move the following 7 directories from the Downloads folder into the BOINC_dev folder (omit any you don't need):
    c-ares-1.13.0
    curl-7.58.0
    openssl-1.1.0g
    wxWidgets-3.1.0
    freetype-2.9
    ftgl-2.1.3~rc5
    sqlite-3.22.0

and

> [4] Get the BOINC source tree from the repository, and put it in the same BOINC_dev folder.

Note the words **same BOINC_dev folder**

**Comment by CharlieFenton** [comment]
Perhaps this needs to be emphasized in the _HowToBuildBOINC_XCode.rtf_ and _HowToBuildBOINC_XCode.pdf_ documents by putting those words in boldface type.

Incidentally, the way I have been updating this document is to edit the RTF file in the _mac_build_ folder, then create a PDF from it and finally creating a single pull request for both files. The PDF file is needed because browsers typically display PDF files but not RTF files, so the link from https://boinc.berkeley.edu/trac/wiki/MacBuild points to that PDF.

**Comment by ShanghaiTimes** [comment]
Charlie,
thanks for the update. Yes I noted that, but I guess what I didn’t say and should have done is for users to note that if you have a specific GitHub directory where you keep repos, then you need to be careful where you put BOINC, or boinc, in that structure, and by extension where you put the external downloaded files so they can be found.

I started by not putting the BOINC_dev structure into my own GitHub repo directory. I cloned it into a directory I called ’sec’. As in Robert/Documents/src/BOINC_dev/boinc … In fact, my GitHub folder structure is different and when I moved the repo to it, initially left the external program folder behind.
My GitHub structure is
Robert/Documents/GitHub/ShanghaiTimes/boinc so I had to move all the downloaded external programs into Robert/Documents/GitHub/ShanghaiTimes

My GitHub.com <http://github.com/> location is ShanghaiTimes/boinc

So my text file addition was trying to remind myself that those eternal programs MUST live in the folder one level above the ‘boinc’ folder.

and I take your point on the rtf file.
Anyway, I do hope I haven't caused any problems.
Robert

**Comment by CharlieFenton** [comment]
> Anyway, I do hope I haven't caused any problems.

Not at all. It's great to get feedback from others, especially about things like build instructions, since different people's experiences will vary. What seems clear to one person may not be clear to another. We very much appreciate your taking the time to share your thoughts.

The reason I explained how to update the  _HowToBuildBOINC_XCode.rtf_ and _HowToBuildBOINC_XCode.pdf_ documents was to encourage you to submit your suggestions for changes to the wording to make things clearer. You can do this as part of this PR or a separate one.

**Comment by TheAspens** [comment]
Greetings - what are the next steps in this PR?  @ShanghaiTimes - I think @CharlieFenton left in in your court to decide if you wanted to do the additional work in this PR or a separate one.  If you could let him know your plan then I think this can move forward. 

Thanks to both of you!

**Comment by ShanghaiTimes** [comment]
Hi Kevin
I read this part as meaning that @CharlieFenton would simply leave it as it is. Or, rewrite the whole thing.

“The reason I explained how to update the  HowToBuildBOINC_XCode.rtf and HowToBuildBOINC_XCode.pdf documents was to encourage you to submit your suggestions for changes to the wording to make things clearer. You can do this as part of this PR or a separate one.”

So I guess I can read it as being either? In any event I seem to recall having made some update sometime back. I’ll have to revisit it to see what was happening.
Let me know anyway if there are other decisions, meantime I’ll have another look at it.

Regards
Robert



__________
Robert Chalmers
https://robert-chalmers.uk
author@robert-chalmers.uk
@R_A_Chalmers

**Comment by TheAspens** [comment]
My bad - I misunderstood the discussion.  Let me try again.  Is @CharlieFenton asking you to take these notes that you have written and instead of adding them as a separate document, to instead incorporate your feedback into HowToBuildBOINC_XCode.rtf and HowToBuildBOINC_XCode.pdf so that those instructions contain what you learned from the process?

If that is correct, then I guess the work needed for this PR is to remove the notes.txt and instead modify the docs directly?

Sorry for the confusion.

**Comment by ShanghaiTimes** [comment]
I didn't want to make any alterations to the original rtf file, and by implication to the pdf file, so I attached the entirety of the "Note to build using XCode 10.2" file I originally sent up, to the bottom of the rtf file.
So - rtf now has that text appended, and I saved it to a pdf file as well. Although if anyone wants to save their own version of the pdf - or rtf for that matter, they should. It would seem that no two versions of a pdf file are the same.

In response @TheAspens (Kevin) yes, I think that's probably right. Charlie's suggestion I mean. So I did. Of course, I can't remove the .txt file at your end.
Hope that's ok.
I'm also still a little unsure of what liberties I can take in these matters.

**Comment by TheAspens** [comment]
@ShanghaiTimes, It is my opinion that you can take all liberties with changes you propose.  The review process before the pull request will allow other people to comment on your changes and if they make sense.  It is better to be bold and integrate your thoughts into the document then to have a lot of caveats at the end that might get missed (note that I have only briefly scanned it so if I misunderstood the change you made just let me know).  Let me know if you want to update your changes based on those comments.

As far as deleting the file, you should be able to run these commands:

```
git rm "mac_build/Note to build using XCode 10.2.txt"
git add . --all
```
and it should remove the file.

**Comment by TheAspens** [comment]
@ShanghaiTimes - if you can run the commands from my previous comment, then I can go ahead and merge.  Let me know if you need any help.  Thanks!

**Comment by ShanghaiTimes** [comment]
> @ShanghaiTimes - if you can run the commands from my previous comment, then I can go ahead and merge. Let me know if you need any help. Thanks!

Ok, I'm a little confused now. I see github/boinc, I have my own login github/shanghaitimes/boinc and on my Desktop I have BoiincDesktop reflecting shanghaitimes/boinc, and running those suggested commands results in:

`zeus:boinc robert$ git rm "mac_build/Note to build using XCode 10.2.txt"
rm 'mac_build/Note to build using XCode 10.2.txt'
zeus:boinc robert$ git add . --all
warning: adding embedded git repository: x-boinc
hint: You've added another git repository inside your current repository.
hint: Clones of the outer repository will not contain the contents of
hint: the embedded repository and will not know how to obtain it.
hint: If you meant to add a submodule, use:
hint: 
hint: 	git submodule add <url> x-boinc
hint: 
hint: If you added this path by mistake, you can remove it from the
hint: index with:
hint: 
hint: 	git rm --cached x-boinc
hint: 
hint: See "git help submodule" for more information.`

**Comment by ShanghaiTimes** [comment]
Hello Kevin
I think I got it this time.

Removed the txt file and synced it. I think.

I have a repo on my desktop that is a copy of my fork on github/ShanghaiTimes, forked from GitHub/boinc

So anyway, I think it is ok now.
Let me know if I need to do anything else.
Regards 
Robert



-----
Robert Chalmers
https://robert-chalmers.uk
author@robert-chalmers.uk
@R_A_Chalmers

**Comment by TheAspens** [comment]
That looks good.  Thanks for the updated documentation!

**Comment by TheAspens** [comment]
I'm closing/opening to retrigger CI so that we can get clean outcome before merging
